osbrain/tests/test_agent_sync_publications.py

```diff
@@ -73,6 +81,173 @@ def on_error(agent):
     agent.error_log.append('error')
 
 
+def match_tail(agent, data):
+    return agent.received[-len(data):] == data
+
+
+@pytest.mark.parametrize('socket_type', ['PUB', 'SYNC_PUB'])
+def test_unsubscribe(nsproxy, socket_type):
+    '''
+    Test that unsubscribing from topics works fine.
+
+    The server will be publishing the natural numbers, in ascending order,
+    starting at one. Odd numbers will be sent with `odd` topic and even
+    numbers with `even` topic.
+
+    A subscriber will subscribe to all numbers, and then it will unsubscribe
+    from the odd numbers.
+
+    Two non consecutive numbers must be received consecutively at that point,
+    since they both will be even.
+    '''
+    def check_non_consecutive(agent):
+        """
+        Check whether the last two received numbers are non-consecutive.
+        """
+        if len(agent.received) < 2:
+            return False
```

Comment:
This (`if len(agent.received) < 2`) seems to never happen. If it does not, then remove it. Otherwise make sure it always happens to ensure maximum coverage of our tests as well. :wink: